### ISSUE TITLE
Internal Server Error (500) on login

### ISSUE THREAD

**Issue by Napsty** [slauger/check_netscaler_gateway#2]
Thanks for the plugin first of all! 

I just tried it on our infra using a NetScaler Virtual Appliance 450010, Build Version: NetScaler NS12.0: Build 57.19.nc. It seems that the default login path (/cgi/login) could not be found:

```
$ /usr/lib/nagios/plugins/check_netscaler_gateway.pl -H login.example.com -u myuser -p mypass -vvv
[extra-opts] check_netscaler_gateway -H login.example.com -u myuser -p mypass -vvv
** POST https://login.example.com/cgi/login ==> 404 Not Found
NetScaler Gateway CRITICAL - request to https://login.example.com/Citrix/StoreWeb/cgi/login failed with HTTP 404
```

Is there a possibility that this could be another URI?

**Comment by Napsty** [comment]
I'm one step closer and the 404 is now gone. The reason for this is that I'm currently working in the internal networks of the company and an internal DNS entry is pointing me to the internal StoreFront server instead of the external Gateway. 

I tried it from the Internet now where DNS is correctly pointing to the gateway. But here I get an error 500:

```
# ./check_netscaler_gateway.pl -H login.example.com -u myuser -p mypass -vvv
[extra-opts] check_netscaler_gateway -H login.example.com -u myuser -p mypass -vvv
** POST https://login.example.com/cgi/login ==> 500 Internal Server Error
NetScaler Gateway CRITICAL - request to https://login.example.com/Citrix/StoreWeb/cgi/login failed with HTTP 500
```

Any idea why the internal server error comes up?

**Comment by slauger** [comment]
Hi,

`login.example.com`needs to point to your external IP address of your NetScaler Gateway vServer. Is this the case? It's the one where the login page to your VPN lives (/vpn/index.html).

**Comment by Napsty** [comment]
> login.example.comneeds to point to your external IP address of your NetScaler Gateway vServer

Yes, that was my mistake. See my second comment. I therefore adjusted the issue title. Now I am sure I am on the correct gateway (yes, where /vpn/index.html lives) and here I get the error 500.

**Comment by slauger** [comment]
HTTP 500 could occur if the login fails for any reason. The reason could be that your credentials are just wrong or you hit a not expected authentication policy.
I would check the aaa.debug log in the first step (`cat /tmp/aaa.debug` on the NetScaler shell). Do you see anything in there?

**Comment by Napsty** [comment]
The output of /tmp/aaad.debug contains some confidential data. I will send it to you in private. 
But it looks like the authentication succeeded, at least from the output.

**Comment by Napsty** [comment]
I was able to fix this by adjusting some of the lwp post commands. PR will follow.

**Comment by Napsty** [comment]
As mentioned in the PR #3 , the plugin now works on our environment after the lwp requests were slightly adjusted.

```
$ /usr/lib/nagios/plugins/check_netscaler_gateway.pl -H login.example.com -u myuser -p mypass -S MyStore  -v
** POST https://login.example.com/cgi/login ==> 302 Object Moved
** POST https://login.example.com/cgi/setclient?wica ==> 200 OK
** POST https://login.example.com/Citrix/MyStoreWeb/Home/Configuration ==> 200 OK (1s)
** POST https://login.example.com/Citrix/MyStoreWeb/Authentication/GetAuthMethods ==> 200 OK
** POST https://login.example.com/Citrix/MyStoreWeb/GatewayAuth/Login ==> 200 OK
** POST https://login.example.com/Citrix/MyStoreWeb/Resources/List ==> 200 OK
** GET https://login.example.com/cgi/logout ==> 200 OK
NetScaler Gateway OK - Desktop-MCP; Desktop-Users;
```

**Comment by Napsty** [comment]
Solved in PR #3 . Thanks!
